core/trino-main/src/main/java/io/trino/execution/buffer/ArbitraryOutputBuffer.java

```diff
@@ -104,12 +104,6 @@ public void addStateChangeListener(StateChangeListener<BufferState> stateChangeL
         stateMachine.addStateChangeListener(stateChangeListener);
     }
 
-    @Override
-    public boolean isFinished()
```

Comment:
The buffer could only be transitioned to the finished state from within. In other words it couldn't fail on it's own. While with the recent changes the buffer is allowed to fail internally (for example when the `ExchangeSink#finish` operation didn't succeed). I decided to simply expose the state instead of introducing more state specific methods (such as `isFinished`, `isAborted`, `isFailed`, etc.)